### DISCUSSION TITLE
Initialize a namelist from a dictionary

### DISCUSSION THREAD

**Discussion by rfkspada** [marshallward/f90nml#157] [Q&A] [answered]
Thanks for this work.

Is there a way to start a namelist from a python dictionary?

**Comment by marshallward** [comment, marked as answer]
Yes in most cases `Namelist(d)` will convert a dict `d` into a Namelist.

There might be some problematic edge cases, but I would give that a try.

**Reply by marshallward** [reply]
For more explicit instructions:
```
>>> from f90nml import Namelist
>>> d = {'grp': {'a': 1, 'b': 2}}
>>> nml = Namelist(d)
>>> print(nml)
&grp
    a = 1
    b = 2
/
```

**Reply by rfkspada** [reply]
Thank you.
